### PR TITLE
Write recipe to compare user and default config - #3065

### PR DESCRIPTION
**Pull request by arturbosch**

### CODE REVIEW

File: docs/_posts/2020-09-27-additional-diff-config-task.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+```kt
+detekt {
+  ...
+  config = files(...)
+  ...
+}
+```
+
+When the file on this location already exists, your configuration won't be overwritten, and the task is a noop.
```
Comment (schalkms) [inline docs/_posts/2020-09-27-additional-diff-config-task.md:20]:
```suggestion
When this file path already exists, your configuration won't be overwritten, and the task is a noop.
```

**Review by schalkms — APPROVED** [review]
Thanks for writing ✍️ this article. It’s a very helpful reference for potential questions concerning this topic.

**Review by cortinico — APPROVED** [review]

**Comment by Tolriq** [comment]
Actually what I do is not diff updated config with prod config but diff default config with previous default config.

This give a precise and easy to follow list of added checks / changed default values that can be applied to prod after.

Diffing default config with prod configuration that may have dozens of already known changes makes this task harder.

**Comment by BraisGabin** [comment]
> Actually what I do is not diff updated config with prod config but diff default config with previous default config.
> 
> This give a precise and easy to follow list of added checks / changed default values that can be applied to prod after.
> 
> Diffing default config with prod configuration that may have dozens of already known changes makes this task harder.

My workflow is similar at yours. And I have an idea but I must say that I don't use it yet:

Create a patch file between the default config and my actual config. This way, the next time that I update detekt I can take the new default and apply the patch to it. And then I can diff my old config with the merge between my customization and the new default.

**Pull request merged**